### ISSUE TITLE
Installed Windows App can't find main Java class

### ISSUE THREAD

**Issue by mcolburn** [javafx-maven-plugin/javafx-maven-plugin#178]
I am able to successfully use the javafx-maven-plugin to create a "native" app for a Mac.  But, not on Windows 7.  The plug-in creates the installation exe, and it runs successfully, installing the app to the user's home directory \AppData\Local\{app name}.  When the installed exe is invoked, an error message displays that "class net/ages/alwb/utils/app/App not found."  When the message is closed, another pops up that says "Failed to launch JVM".  

The jar seems to be fine.  It can be invoked using java -jar {jar file name}.  That finds the main class just fine. The MANIFEST.MF file in the jfx jar for the app contains the correct name of the main class.'

It is only failing when started using the exe launcher.  I also tried adding the jar to the classpath to see if that would fix the problem, but doesn't.

The jvm is not getting the classpath information it needs when the exe launcher is used.  Is there some configuration required in the pom.xml?

Thank you in advance for assistance!

**Comment by FibreFoX** [comment]
Hi @mcolburn , pleasse check which version of javafx-maven-plugin you are using and what version of JDK you are using under windows 7.

>  It can be invoked using java -jar {jar file name}. 

Do you mean the generated jfx-jar, or the generated jar by maven at the destination `/target/yourProject.jar` ?

**Comment by mcolburn** [comment]
javafx-maven-plugin: version 8.1.4
jdk: 1.8.0_65

Both the jfx-jar and the target/myproject.jar run fine using java -jar {jarfile name}

It only fails to find the main app when invoked via the exe

Thanks for your assistance!

**Comment by FibreFoX** [comment]
@mcolburn please update to javafx-maven-plugin 8.2.0, there are some workarounds for native launchers I had to implement. More see the [changelogs](https://github.com/javafx-maven-plugin/javafx-maven-plugin/blob/master/CHANGELOG.md)

**Comment by FibreFoX** [comment]
I assume you are hitting the "there is a dot in my application-name"-bug :smile:

**Comment by FibreFoX** [comment]
@mcolburn I can see you pushed the app in question to some git-repository: https://github.com/AGES-Initiatives/alwb-utils-gui/blob/master/alwb.utils/pom.xml

Does it work with the newer version?

**Comment by mcolburn** [comment]
Sorry…no, I just now tried it, and it did not solve the problem.  Same results.

**Comment by FibreFoX** [comment]
Was my assumption correct that the problem is that alwb.utils-application? Where can I find some code to reproduce? (I would try to reproduce it with `java version "1.8.0_60"` on windows 8.1)

**Comment by russ-p** [comment]
I have the save issue if  username contains non latin characters (C:/Users/Пользователь/AppData/…)
What is your username?

**Comment by mcolburn** [comment]
Here is the Github repo for the application: https://github.com/AGES-Initiatives/alwb-utils-gui <https://github.com/AGES-Initiatives/alwb-utils-gui>

**Comment by FibreFoX** [comment]
As far as I can see, there is nothing wrong with the javafx-maven-plugin, because everything is bundled as requested and expected.

Your application is the problem itself ;) after disabling the stuff inside `net.ages.alwb.utils.app.App::start` by replacing it with just showing the stage, I got an exception while running the generated jfx-jar, caused by the path inside `src/main/resources/log4j.properties` (by calling it via `java -jar`, not by the native launcher), you should change this to a non-absolute path.

I'm still stuck to understand what happens ... will report back when i found something

**Comment by FibreFoX** [comment]
@russ-p its not because of the path, the problem already exists when calling `target/jfx/native/alwb.utils-v1.0/alwb.utils-v1.0.exe` inside normal folder

**Comment by mcolburn** [comment]
Danny,

If you are able to start the app, then you are not experiencing the issue, despite the problems you ran into.  In other words, you are getting beyond the problem I have reported.

My issue is that the exe launcher reports that the main class path is not found.

Later I will make a hello world app and see if I can duplicate the issue. I will let you know either way.

Yes—sorry about the log4j path—the code is not releasable yet.

Thanks!

**Comment by FibreFoX** [comment]
@mcolburn No problem :) i suspect there is something wrong with the classes inside the generated jfx-jar AND the jars inside the lib-folder, something like "doublicate class problem" resulting in disposing the jfx-jar. Like while putting some javaee-lib into EE-containers ;)

The best would be to strip your project down to some reproducable minimal application. There is a very lot inside your sources-folder, so maybe the mix is confusing the native launcher (but don't know why yet).

**Comment by mcolburn** [comment]
A simple helloworld javafx app works fine with Windows 7 using javafx-maven-plugin 8.2.0.

**Comment by FibreFoX** [comment]
@mcolburn In that case please try to debug your application a bit more.

Until you found something out, I will close this issue as invalid.

**Comment by mcolburn** [comment]
That is fine.  Thank you for the great work you are doing and your willing support for issues!

**Comment by kanikachauhan** [comment]
Hi,
I am facing the same issue. Once,the installer is created and it is successfully installed, it shows main not found error

**Comment by FibreFoX** [comment]
@kanikachauhan can you provide some reproducable project for this?

**Comment by kanikachauhan** [comment]
can i upload on github and share the link?

**Comment by FibreFoX** [comment]
@kanikachauhan does not matter :) as long as I can get the project will be sufficient

**Comment by philippbc** [comment]
I ran into the same problem. Uninstalling my app before updating fixed the problem for me.

**Comment by randywallace** [comment]
I ran into this one in a very hard way.  I had created a logback appender that would write logging to a TextArea.  When I added the appender to logback.xml, this happened (weird logback classloading magic).  When I created a static class that configured logback in code and loaded it in the main function, kablammo.  Only until I added the appender in the JavaFX start() (or presumeably afterward) did things cooperate.  Of course, this only happened when I installed and ran a native build.

Lesson learned, right?  https://stackoverflow.com/questions/25070863/how-does-javafx-main-method-launchargs-work

Now, that being said, the Inno EXE does call main (there's something else I call in there that is working which I should rip out), but clearly there's some other magic that happens in there I will never know.
